core/src/main/java/com/linecorp/armeria/common/logging/ContentPreviewerFactory.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.logging;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiPredicate;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestContext;
+
+@FunctionalInterface
+public interface ContentPreviewerFactory {
+    ContentPreviewer get(RequestContext ctx, HttpHeaders headers);
+
+    default ContentPreviewerFactory asImmutable() {
+        return this;
+    }
+
+    ContentPreviewerFactory DISABLED = (ctx, headers) -> ContentPreviewer.DISABLED;
+
+    /**
+     * TODO: AddJavadocs.
+     */
+    static ContentPreviewerFactory of(MediaType contentType, Supplier<ContentPreviewer> supplier) {
+        return of((ctx, headers) -> {
+            final MediaType type = headers.contentType();
+            return type != null && type.is(contentType);
+        }, supplier);
+    }
+
+    static ContentPreviewerFactory of(String contentType, Supplier<ContentPreviewer> supplier) {
+        return of(MediaType.parse(contentType), supplier);
+    }
+
+    /**
+     * TODO: AddJavadocs.
+     */
+    static ContentPreviewerFactory of(BiPredicate<RequestContext, HttpHeaders> predicate,
+                                      Supplier<ContentPreviewer> supplier) {
+        return (ctx, headers) -> {
+            if (predicate.test(ctx, headers)) {
+                return supplier.get();
+            }
+            return ContentPreviewer.DISABLED;
+        };
+    }
+
+    /**
+     * TODO: AddJavadocs.
+     */
+    static ContentPreviewerFactory of(ContentPreviewerFactory factory1, ContentPreviewerFactory factory2) {
```

Comment:
This method could be removed because `of(ContentPreviewerFactory...)` will handle this case as well.